src/pages/account/settings/model.ts

```diff
@@ -0,0 +1,119 @@
+import { AnyAction, Reducer } from 'redux';
+import { EffectsCommandMap } from 'dva';
+import { CurrentUser, GeographicItemType } from './data.d';
+import { queryCity, queryCurrent, queryProvince, query as queryUsers } from './service';
+
+export interface ModalState {
+  currentUser?: Partial<CurrentUser>;
+  province?: GeographicItemType[];
+  city?: GeographicItemType[];
+  isLoading?: boolean;
+}
+
+export type Effect = (
+  action: AnyAction,
+  effects: EffectsCommandMap & { select: <T>(func: (state: ModalState) => T) => T },
+) => void;
+
+export interface ModelType {
+  namespace: string;
+  state: ModalState;
+  effects: {
+    fetchCurrent: Effect;
+    fetch: Effect;
+    fetchProvince: Effect;
+    fetchCity: Effect;
+  };
+  reducers: {
+    saveCurrentUser: Reducer<ModalState>;
+    changeNotifyCount: Reducer<ModalState>;
+    setProvince: Reducer<ModalState>;
+    setCity: Reducer<ModalState>;
+    changeLoading: Reducer<ModalState>;
+  };
+}
+
+const Model: ModelType = {
+  namespace: 'accountAndsettings',
+
+  state: {
+    currentUser: {},
+    province: [],
+    city: [],
+    isLoading: false,
+  },
+
+  effects: {
+    *fetch(_, { call, put }) {
+      const response = yield call(queryUsers);
+      yield put({
+        type: 'save',
+        payload: response,
+      });
+    },
+    *fetchCurrent(_, { call, put }) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [expected parameter: '{ call, put }' to have a typedef](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4583897)